lib/spack/spack/spec.py

```diff
@@ -3514,76 +3656,72 @@ def satisfies(self, other, deps=True, strict=False):
         ):
             return False
         if self.versions and other.versions:
-            if not self.versions.satisfies(other.versions, strict=strict):
+            if not self.versions.satisfies(other.versions):
                 return False
-        elif strict and (self.versions or other.versions):
+        elif self.versions or other.versions:
             return False
 
         # None indicates no constraints when not strict.
```

Comment:
more strict remnants.

Reply:
These should have been removed. Let me know if there are others